tests/ui/let_if_seq.stderr

```diff
@@ -1,3 +1,23 @@
+error: unneeded late initalization
+  --> $DIR/let_if_seq.rs:48:5
+   |
+LL |     let foo;
+   |     ^^^^^^^^
+   |
+   = note: `-D clippy::needless-late-init` implied by `-D warnings`
+help: declare `foo` here
+   |
+LL |     let foo = if f() {
+   |     +++++++++
+help: remove the assignments from the branches
+   |
+LL |         0
+   |
+help: add a semicolon after the if expression
+   |
+LL |     };
+   |      +
+
```

Comment:
This overlaps a little with `useless_let_if_seq`, `needless_late_init` doesn't do as many clever transformations as `useless_let_if_seq` however, hopefully meaning it has less false positives

I wasn't sure what to choose here, I could remove the matching of ifs from `needless_late_init`

Reply:
> This overlaps a little with `useless_let_if_seq`, needless_late_init doesn't do as many clever transformations as `useless_let_if_seq` however, hopefully meaning it has less false positives

Yes, it looks good.

> I wasn't sure what to choose here, I could remove the matching of ifs from `needless_late_init`

IMO it looks good as it is.